network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/revoked/BetaEndStore.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.networkprotection.impl.revoked
+
+import android.content.SharedPreferences
+import androidx.core.content.edit
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.mobile.android.vpn.prefs.VpnSharedPreferencesProvider
+import com.squareup.anvil.annotations.ContributesBinding
+import javax.inject.Inject
+
+interface BetaEndStore {
+    fun showBetaEndDialog()
+    fun betaEndDialogShown(): Boolean
+}
```

Comment:
Can we make this entire class an implementation detail of `BetaEndedDialog`. There is a lot of logic that is outsdie `BetaEndedDialog` that in reality should be impl detail (will leave another comment) about this

Reply:
Done